Autopilot/boardfiles/Src/freertos.c

```diff
@@ -136,40 +160,143 @@ void MX_FREERTOS_Init(void) {
+  sensorFusionHandle = osThreadCreate(osThread(sensorFusionRun), NULL);
+
+  osThreadDef(testThread, testExecute, osPriorityNormal, 0, 128);
+  test = osThreadCreate(osThread(testThread), NULL);
+
+
+  /* definition and creation of sensorFusionRun */
 
   /* USER CODE BEGIN RTOS_THREADS */
   /* add threads, ... */
   /* USER CODE END RTOS_THREADS */
 
 }
 
-/* USER CODE BEGIN Header_StartDefaultTask */
+/* USER CODE BEGIN Header_attitudeManagerExecute */
 /**
-  * @brief  Function implementing the defaultTask thread.
+  * @brief  Function implementing the attitudeManager thread.
   * @param  argument: Not used
   * @retval None
   */
-/* USER CODE END Header_StartDefaultTask */
-void StartDefaultTask(void const * argument)
+/* USER CODE END Header_attitudeManagerExecute */
+void attitudeManagerExecute(void const * argument)
+{
+  /* USER CODE BEGIN attitudeManagerExecute */
+  /* Infinite loop */
+  while(1)
+  {
+    TickType_t xLastWakeTime = xTaskGetTickCount();
+    vTaskDelayUntil(&xLastWakeTime, PERIOD_ATTITUDEMANAGER_MS);
+    bool status = AttitudeManagerInterfaceExecute();
+    if (!status) {
+      catastrophicFailure = true;
+    }
+    HAL_GPIO_TogglePin(LED1_GPIO_Port, LED1_Pin);
+  }
+  /* USER CODE END attitudeManagerExecute */
+}
+
+/* USER CODE BEGIN Header_pathManagerExecute */
+/**
+* @brief Function implementing the pathManager thread.
+* @param argument: Not used
+* @retval None
+*/
+/* USER CODE END Header_pathManagerExecute */
+void pathManagerExecute(void const * argument)
+{
+  /* USER CODE BEGIN pathManagerExecute */
+  /* Infinite loop */
+  while(1)
+  {
+    TickType_t xLastWakeTime = xTaskGetTickCount();
+    vTaskDelayUntil(&xLastWakeTime, PERIOD_PATHMANAGER_MS);
+    bool status = PathManagerInterfaceExecute();
+    if (!status) {
+      catastrophicFailure = true;
+    }
+    HAL_GPIO_TogglePin(LED2_GPIO_Port, LED2_Pin);
+  }
+  
+  /* USER CODE END pathManagerExecute */
+}
+
+/* USER CODE BEGIN Header_StartTelemetryRun */
+/**
+* @brief Function implementing the telemetryRun thread.
+* @param argument: Not used
+* @retval None
+*/
+/* USER CODE END Header_StartTelemetryRun */
+void StartTelemetryRun(void const * argument)
 {
-  /* USER CODE BEGIN StartDefaultTask */
+  /* USER CODE BEGIN StartTelemetryRun */
   /* Infinite loop */
-  for(;;)
+  while(1)
   {
-    osDelay(1);
+    TickType_t xLastWakeTime = xTaskGetTickCount();
+    vTaskDelayUntil(&xLastWakeTime, PERIOD_TELEMETRY_MS);
+    bool status = TelemetryManagerInterfaceExecute();
+    if (!status) {
+      catastrophicFailure = true;
+    }
+    HAL_GPIO_TogglePin(LED3_GPIO_Port, LED3_Pin);
+  }
+  
+  /* USER CODE END StartTelemetryRun */
+}
+
+void SensorFusionExecute(void const * argument) {
+  /* USER CODE BEGIN SensorFusionExecute */
+  /* Infinite loop */
+
+  while (1) {
+    TickType_t xLastWakeTime = xTaskGetTickCount();
+    vTaskDelayUntil(&xLastWakeTime, PERIOD_SENSORFUSION_MS);
+    SFError_t err = SensorFusionInterfaceExecute();
```

Comment:
Is there an error case we want to consider?

Reply:
I'll look into it, I think if there is an error I'll just set catastrophicError to true similar to the other threads